### PR TITLE
feat: add 'show connector plugins' syntax

### PR DESCRIPTION
**Pull request by spena**
### Description 
_What behavior do you want to change, why, how does your patch achieve the changes?_
Feature https://github.com/confluentinc/ksql/issues/4623

Add new `SHOW CONNECTOR PLUGINS` syntax that displays the all available connector plugins from Connect.
```
ksql> show connector plugins;

 Class                                                     | Type   | Version    
---------------------------------------------------------------------------------
 io.confluent.connect.replicator.ReplicatorSourceConnector | SOURCE | 6.1.0-0    
 org.apache.kafka.connect.file.FileStreamSinkConnector     | SINK   | 6.1.0-0-ce 
 org.apache.kafka.connect.file.FileStreamSourceConnector   | SOURCE | 6.1.0-0-ce 
 org.apache.kafka.connect.mirror.MirrorCheckpointConnector | SOURCE | 1          
 org.apache.kafka.connect.mirror.MirrorHeartbeatConnector  | SOURCE | 1          
 org.apache.kafka.connect.mirror.MirrorSourceConnector     | SOURCE | 1          
---------------------------------------------------------------------------------
```

### Testing done 
_Describe the testing strategy. Unit and integration tests are expected for any behavior changes._
Added unit tests
Verified manually in CLI

### Reviewer checklist
- [ ] Ensure docs are updated if necessary. (eg. if a user visible feature is being added or changed).
- [ ] Ensure relevant issues are linked (description should include text like "Fixes #<issue number>")

### CODE REVIEW

File: ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/execution/ListConnectorPluginsExecutor.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+          info.className(),
+          info.type(),
+          info.version()
+      ));
+    }
+
+    return Optional.of(
+      new ConnectorPluginsList(
+        configuredStatement.getStatementText(),
+         warnings,
```
Comment (jzaralim) [inline ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/execution/ListConnectorPluginsExecutor.java:68]:
```suggestion
        warnings,
```

File: ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/execution/ListConnectorPluginsExecutor.java [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+        serviceContext.getConnectClient().connectorPlugins();
+    if (plugins.error().isPresent()) {
+      return Optional.of(new ErrorEntity(
+        configuredStatement.getStatementText(),
+        plugins.error().get()
+      ));
+    }
+
+    final List<SimpleConnectorPluginInfo> pluginInfos = new ArrayList<>();
+    final List<KsqlWarning> warnings = new ArrayList<>();
```
Comment (jzaralim) [inline ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/execution/ListConnectorPluginsExecutor.java:55]: Is this supposed to be an empty list? If so, could we remove `warnings` from `ConnectorPluginsList`?

File: ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/execution/ListConnectorPluginsExecutor.java (hunk 1 above)
Comment (spena) [reply ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/execution/ListConnectorPluginsExecutor.java]: Done

File: ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/execution/ListConnectorPluginsExecutor.java (hunk 2 above)
Comment (spena) [reply ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/execution/ListConnectorPluginsExecutor.java]: Done

**Review by jzaralim — APPROVED** [review]
lgtm

**Pull request merged**